src/data.ts

```diff
@@ -58,11 +60,22 @@ export interface Data {
    * A URL from which to load the data set. Use the format.type property
    * to ensure the loaded data is correctly parsed.
    */
-  url?: string;
+  url: string;
+}
+
+export interface InlineData {
   /**
    * Pass array of objects instead of a url to a file.
    */
-  values?: any[];
+  values: any[];
+}
+
+export function isUrlData(data: Data): data is UrlData {
```

Comment:
@NeelMohapatra  If you haven't, you might wanna learn about what is a type guard in typescript.  https://www.typescriptlang.org/docs/handbook/advanced-types.html 

For example, try pulling this branch and change this to `boolean` and see what errors you've got. :)